src/RedirectRequest.jl

```diff
@@ -0,0 +1,57 @@
+module RedirectRequest
+
+import ..Layer, ..request
+using ..URIs
+using ..Messages
+using ..Pairs: setkv
+using ..Header
+import ..@debug, ..DEBUG_LEVEL
+
+
+"""
+    request(RedirectLayer, method, ::URI, headers, body) -> HTTP.Response
+
+Redirects the request in the case of 3xx response status.
+"""
+
+abstract type RedirectLayer{Next <: Layer} <: Layer end
+export RedirectLayer
+
+function request(::Type{RedirectLayer{Next}},
+                 method::String, uri::URI, headers, body;
+                 redirect_limit=3, forwardheaders=false, kw...) where Next
+    count = 0
+    while true
+    
+        res = request(Next, method, uri, headers, body; kw...)
+
+        if (count == redirect_limit
+        ||  !isredirect(res)
+        ||  (location = header(res, "Location")) == ""
+        ||  method == "HEAD") #FIXME why not redirect HEAD?
```

Comment:
yeah, sounds like we should remove this check: https://stackoverflow.com/questions/3820663/is-it-possible-for-xhr-head-requests-to-not-follow-redirects-301-302

Reply:
Hmmm, I'm not sure what XMLHttpRequest is, or if it is relevant to us.

I'm inclined to say at this point that we should not redirect on HEAD requests. Ignoring the semantic content of the header for a HEAD request seems consistent with ignoring Content-Length: https://github.com/samoconnor/HTTP.jl/blob/simplify_parser_branch/src/Messages.jl#L451

Perhaps we should have a `redirect_head::Bool` kw arg like this: https://stackoverflow.com/a/9967683

Reply:
`XMLHttpRequest` is HTTP.jl for javascript; the standard implementation of the http spec for basically the entire web. I don't think it matters all that much either way we go (HEAD requests are not that common anyway), but from that SO, it sounded liked the spec recommends _always_ following HEAD requests automatically.